### DISCUSSION TITLE
Does thorvg support drawing Mosaic effects

### DISCUSSION THREAD

**Discussion by xland** [thorvg/thorvg#1813] [Q&A] [answered]
I developed a [screenshot software](https://github.com/xland/ScreenCapture)
It allows users to draw Mosaic patterns with the mouse.
I am considering using thorvg as the graphics engine for this software.
Does thorvg support this kind of functionality?
Can thorvg do this easily?
Thanks very much.
![image](https://github.com/thorvg/thorvg/assets/2980915/e8f9c94f-5e1c-4126-a385-cb9e1b1e28c7)

**Comment by hermet** [comment, marked as answer]
@xland Hello, currently, ThorVG doesn't support built-in mosaic effects. You will need to implement your own mosaic pattern logic based on the screenshot and user input. Once it's done, you will be able to draw mosaic effects, as ThorVG supports colored-tile or triangle drawings.

**Comment by xland** [comment]
Thanks for your reply.
I want to keep this question for now.
I think I'll ask you again after I finish my demo code.
